### PR TITLE
Functional Test

### PR DESCRIPTION
**Pull request by stephenrlouie**
### Change log

First pass on functional tests, to be deployed in lab

### Notes

1. Stands up a couchbase container
2. Runs sigma from the latest container
3. Adds the `example/hello-world` plugin
4. Adds two hello services, the second uses the output of the first as an input.
5. Validates the output

We could technically run this in travis since its a single machine test that depends on containers.

### TODO

- [x] validate and set up system in lab
- [ ] ~possibly plug into travis?~

### CODE REVIEW

File: tests/functional/hello-world/hello_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+func postJSONFile(urlPath string, filePath string) (*http.Response, error) {
+	b, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	return http.Post(makeUrl(urlPath), "application/json", bytes.NewBuffer(b))
+}
+
+//Currenty waits 10 seconds for the API to be reachable
```
Comment (mikeynap) [inline tests/functional/hello-world/hello_test.go]: Fake Comment: It seems to currently wait 30 seconds. Sad. Lamestream Commenter.

File: Makefile [hunk 2]
```diff
@@ -14,9 +16,14 @@ test:
 
 integration:
 	./tests/integration/setup.sh
-	-go test -v  --ldflags '${EXTLDFLAGS}' $(PACKAGES_FOR_INTEGRATION)
+	go test -v  --ldflags '${EXTLDFLAGS}' $(PACKAGES_FOR_INTEGRATION)
 	./tests/integration/shutdown.sh
 
+functional:
+	./tests/functional/setup.sh
+	go test -v  --ldflags '${EXTLDFLAGS}' $(PACKAGES_FOR_FUNCTIONAL)
```
Comment (mikeynap) [inline Makefile]: why remove the -? I thought it made sure the next line ran.

File: tests/functional/hello-world/hello_test.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+			decode := json.NewDecoder(resp.Body)
+			if err = decode.Decode(&sigmaS); err != nil {
+				return err
+			}
+
+			// Checks if the output matches what we expect
+			if resp != nil && resp.StatusCode != http.StatusOK {
+				return fmt.Errorf("Error AddingPlugin. Received Status Code: %d", resp.StatusCode)
+			} else {
+				if sigmaS.Output != nil {
```
Comment (mikeynap) [inline tests/functional/hello-world/hello_test.go]: Shuld b: `if len(sigmaS.Output) > 0`

File: Makefile (hunk 2 above)
Comment (stephenrlouie) [inline Makefile]: my b

**Pull request merged**